ulib/FStar.Pervasives.fsti

```diff
@@ -1125,6 +1125,20 @@ val bind_has_range_args : unit
   *)
 val primitive_extraction : unit
 
+(** A qualifier on a type definition which when used in co-domain position
+    on an arrow type will be extracted as if it were an impure effect type.
+
+    e.g., if you have
+    
+    [@@extract_as_impure_effect]
+    val stt (a:Type) (pre:_) (post:_) : Type
+
+    then arrows of the form `a -> stt b p q` will be extracted 
+    similarly to `a -> Dv b`.
+ *)
+val extract_as_impure_effect : unit
```

Comment:
I wonder if we should check this attribute, e.g., when typechecking a let definition, if it has the attribute, then allow only arrows whose codomain is impure effect?

Reply:
Claiming that something is PURE is risky and would require checking. But, claiming that it is IMPURE is conservative, no?